x-pack/plugins/ml/public/application/services/dashboard_service.ts

```diff
@@ -27,9 +27,18 @@ export function dashboardServiceProvider(dashboardService: DashboardStart) {
       return responses.hits;
     },
     /**
-     * Generates dashboard url with edit mode
+     * Fetch dashboards by id
      */
-    async getDashboardEditUrl(dashboardId: string) {
+    async fetchDashboardsById(ids: string[]) {
+      const findDashboardsService = await dashboardService.findDashboardsService();
+      const responses = await findDashboardsService.findByIds(ids);
```

Comment:
I'm seeing 404 errors from this call in the console.
<img width="705" alt="image" src="https://github.com/user-attachments/assets/3304ca42-fa98-4764-8c22-bb25fa099c61">

I think we should catch these and just return an empty array if they can't be found.

Reply:
These errors are actually not thrown from here - `findByIds` has it's own error handling internally in `src/plugins/dashboard/public/services/dashboard_content_management/lib/find_dashboards.ts` and returns the caught error in an object with a 'status' and 'error' property. The response is type `FindDashboardsByIdResponse` - which is defined in that same file I shared.

I think if we want to do something to remove the logged 404s, it will need to be in a separate PR and would need to chat with whoever maintains that service.